components/script/dom/xmlhttprequest.rs

```diff
@@ -368,6 +373,11 @@ impl XMLHttpRequestMethods for XMLHttpRequest {
                 // Step 11 - abort existing requests
                 self.terminate_ongoing_fetch();
 
+                // See WPT test: FileAPI/blob/Blob-XHR-revoke.html
```

Comment:
So this is wrong. I think you're aware of this since you've marked it as a hack 😄 

The correct behavior is that XHR queries for validity during `open()`, and is allowed to access the blob even after the URL validity expires, _provided that it was allowed to during `open()`_. Firefox returns an error on invalid blob URIs in open() (for both existing-but-revoked URIs and nonexistant URIs)

We should have a second refcount, which counts the number of XHRs holding on to this. When xhr calls open(), it should send a special message that checks validity and bumps this refcount if still valid; and the message will respond with the validity state (which is now just a bool). When the XHR gets a response or error, it un-bumps the refcount. Deleting an entry from the store is done when the refcounts and validity state are null. HTTP fetches will not fetch URLs with a false validity. New XHR requests will also fail on `open()` for invalid URIs. Existing XHRs will work fine.

The spec says nothing about this, however 😄 

_for now_, don't include this hack, have a simple validity state and let the test fail. We'll work on clearing up the spec and only then implement it.